apis/externalsecrets/v1beta1/clusterexternalsecret_types.go

```diff
@@ -33,9 +33,14 @@ type ClusterExternalSecretSpec struct {
 	ExternalSecretMetadata ExternalSecretMetadata `json:"externalSecretMetadata,omitempty"`
 
 	// The labels to select by to find the Namespaces to create the ExternalSecrets in.
+	// Deprecated: Use NamespaceSelectors instead.
 	// +optional
 	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
 
+	// A list of labels to select by to find the Namespaces to create the ExternalSecrets in. The selectors are ORed.
+	// +optional
+	NamespaceSelectors []*metav1.LabelSelector `json:"namespaceSelectors,omitempty"`
+
 	// Choose namespaces by name. This field is ORed with anything that NamespaceSelector ends up choosing.
 	// +optional
 	Namespaces []string `json:"namespaces,omitempty"`
```

Comment:
I believe we actually don't need the `Namespaces` field anymore since that is covered by `kubernetes.io/metadata.name` label.